apps/challenges/views.py

```diff
@@ -2239,3 +2245,292 @@ def get_challenge_evaluation_cluster_details(request, challenge_pk):
     )
     response_data = serializer.data
     return Response(response_data, status=status.HTTP_200_OK)
+
+
+@api_view(["POST"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def validate_challenge_config(request, challenge_host_team_pk):
+    challenge_host_team = get_challenge_host_team_model(challenge_host_team_pk)
+    response_data = {}
+    serializer = ChallengeConfigSerializer(
+        data=request.data, context={"request": request}
+    )
+    if serializer.is_valid():
+        serializer.save()
+        uploaded_zip_file_path = serializer.data["zip_configuration"]
+    else:
+        response_data["error"] = serializer.errors
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
>    challenge_host_team = get_challenge_host_team_model(challenge_host_team_pk)
    response_data = {}
    serializer = ChallengeConfigSerializer(
        data=request.data, context={"request": request}
    )
    if serializer.is_valid():
        serializer.save()
        uploaded_zip_file_path = serializer.data["zip_configuration"]
    else:
        response_data["error"] = serializer.errors
        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)

Why are we saving the config file when someone is trying to only validate the config?

Reply:
This is because when we get the uploaded zip file. It is a in memory zip file. I will check if there's a way to not save it and directly access.